File: aggregator/src/main/java/com/web3analytics/models/MarketTrend.java
package com.web3analytics.models;

/**
 * Rolling market trend computed from sliding windows over swap events.
 * Captures price momentum, volume trends, and market direction.
 */
public record MarketTrend(
        String windowId,
        long windowStart,
        long windowEnd,
        String pairAddress,
        String token0Symbol,
        String token1Symbol,
        double avgPrice,
        double openPrice,
        double closePrice,
        double priceChangePercent,
        Double volumeUSD,
        int swapCount,
        int uniqueTraders,
        double volatility,
        String trend,
        long processedAt
) {}

File: analytics-service/src/main/kotlin/com/web3analytics/analytics/model/MarketTrend.kt
package com.web3analytics.analytics.model

import com.fasterxml.jackson.annotation.JsonIgnoreProperties

@JsonIgnoreProperties(ignoreUnknown = true)
data class MarketTrend(
    val windowId: String,
    val windowStart: Long = 0,
    val windowEnd: Long = 0,
    val pairAddress: PairAddress,
    val token0Symbol: String? = null,
    val token1Symbol: String? = null,
    val avgPrice: Double = 0.0,
    val openPrice: Double = 0.0,
    val closePrice: Double = 0.0,
    val priceChangePercent: Double = 0.0,
    val volumeUSD: Double? = null,
    val swapCount: Int = 0,
    val uniqueTraders: Int = 0,
    val volatility: Double = 0.0,
    val trend: String = "NEUTRAL",
    val processedAt: Long = 0
)
